File: GrifballWebApp.Server/Teams/CaptainPlacementDto.cs
namespace GrifballWebApp.Server.Teams;

public class CaptainPlacementDto
{
    public int SeasonID { get; set; }
    public int PersonID { get; set; }
    public int OrderNumber { get; set; }
}

public record CaptainAddedDto
{
    public required int SeasonID { get; set; }
    public required int PersonID { get; set; }
    public required string TeamName { get; set; }
    public required string CaptainName { get; set; }
    public required int OrderNumber { get; set; }
}

public class RemoveCaptainDto
{
    public int SeasonID { get; set; }
    public int PersonID { get; set; }
}

File: grifballwebapp.client/src/app/api/dtos/captainPlacementDto.ts
export class CaptainPlacementDto {
  seasonID: number = 0;
  personID: number = 0;
  orderNumber: number = 0;
}

export class CaptainAddedDto {
  seasonID: number = 0;
  personID: number = 0;
  teamName: string = '';
  captainName: string = '';
  orderNumber: number = 0;
}

export class RemoveCaptainDto {
  seasonID: number = 0;
  personID: number = 0;
}
